OpenUpSA/wazimap-ng-ui
# hiding subcategories in data mapper if all the children indicators are qualitative

## Description
in the data mapper we don't show qualitative indicators. but we used to show the subcategories that have only qualitative indicators.

the scenario : 
* there is a subcategory A. 
* subcategory A has 2 children
      - X
      - Y
* both X and Y are qualitative so they are not shown in the data mapper
* expected : subcategory A is not shown too since it has no children indicator available to be shown in the data mapper
* the bug : subcategory A is shown but does not expand
* the fix : subcategory A is not shown in the data mapper


## Related Issue
https://wazimap.atlassian.net/browse/WNCM-112

## How is it tested automatically?
added a unit test in `__tests__/setup/dataexplorer.test.js`

## How should a reviewer test it locally
* confirm that in the data mapper we do not show the qualitative indicators
* confirm that in the data mapper we do not show the subcategories that have only qualitative indicators
* an example subcategory that has only qualitative indicators
      - go to SANEF profile
      - set the API url staging
      - navigate to KwaZulu-Natal > Uthukela District > Alfred Duma
      - in the data mapper Elections > 2021 Municipal elections subcategory has only qualitative indicators so should not be shown 


## Screenshots
before :
![image](https://user-images.githubusercontent.com/53019884/140825234-4d9c930c-d2c8-45fe-8298-071bef787a46.png)

after :
![image](https://user-images.githubusercontent.com/53019884/140825335-198447ab-31da-4968-836b-60d7db7488ff.png)


## Changelog

### Added

### Updated
* `src/js/utils.js` to check for the `dataset_content_type` while deciding if a subcategory has children or not

### Removed


## Checklist


### Pull Request


### Code Quality


### Testing
